instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -141,12 +166,23 @@ def process_request(self, request):
         if self._excluded_urls.url_disabled(request.build_absolute_uri("?")):
             return
 
+        is_asgi_request = _is_asgi_request(request)
+        if is_asgi_request and not _is_asgi_supported:
+            return
```

Comment:
> Can a django service/middleware handle both WSGI and ASGI requests at the same time?
Yes, it can handle both, so we need to check which kind of request we're receiving on each call.

> At least is_asgi_supported can be called before _is_asgi_request and that way we can avoid running the type check for non-ASGI requests.
As we're using `is_asgi_request` multiple times on this method, the code will need to be a bit more convoluted if we want to avoid the type check when we know ASGI is not supported. I will change the code accordingly, and we can decide if it's worth to keep it that way.